### PR TITLE
Model for UndeliverableMessage

### PR DESCRIPTION
**Pull request by tayloraburgess**
- Add some additional model fields to the one we whiteboarded that are available on the message and might be useful (the topic arn & the initial message timestamp)
- `test_settings.py` is modeled off the same in atomiq

### CODE REVIEW

File: mbq/pubsub/models.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+class UndeliverableMessage(models.Model):
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    type = models.CharField(max_length=256)
+    payload = models.TextField()
+    timestamp = models.DateTimeField()
+    queue = models.CharField(max_length=80)
+    topic_arn = models.CharField(max_length=2048)
+
+    error_timestamp = models.DateTimeField()
+    error = models.TextField()
```
Comment (Pastromhaug) [inline mbq/pubsub/models.py]: in Atomiq we have `error_message` and `stacktrace` as two separate fields. This allowed us to display the error message in the admin list view, and then the full stack trace when you click into the task. Could be useful to do the same here.

File: mbq/pubsub/models.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+class UndeliverableMessage(models.Model):
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    type = models.CharField(max_length=256)
+    payload = models.TextField()
+    timestamp = models.DateTimeField()
+    queue = models.CharField(max_length=80)
+    topic_arn = models.CharField(max_length=2048)
+
+    error_timestamp = models.DateTimeField()
```
Comment (Pastromhaug) [inline mbq/pubsub/models.py]: What is the difference between `error_timestamp` and `timestamp`?

File: mbq/pubsub/models.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import uuid
+
+from django.db import models
+
+
+class UndeliverableMessage(models.Model):
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    type = models.CharField(max_length=256)
+    payload = models.TextField()
+    timestamp = models.DateTimeField()
```
Comment (Pastromhaug) [inline mbq/pubsub/models.py]: not sure what timestamp is, but on most models we conventionally use `created_at` as the name of the field which stores the time at which the row was inserted.

File: mbq/pubsub/models.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import uuid
+
+from django.db import models
+
+
+class UndeliverableMessage(models.Model):
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    type = models.CharField(max_length=256)
```
Comment (Pastromhaug) [inline mbq/pubsub/models.py]: personally I would make this `message_type` for clarity, even though technically it is redundant. Up to you

Comment (mattbriancon) [reply mbq/pubsub/models.py]: +1. `type` is a reserved word in Python and is usually overly broad.

File: tests/test_settings.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import os
```
Comment (mattbriancon) [inline tests/test_settings.py]: I have the settings in the pytest config if you want to consolidate.

File: requirements-dev.txt [hunk 6]
```diff
@@ -10,3 +10,6 @@ pytest-django
 tox
 tox-travis
 tox-venv
+psycopg2-binary
+dj_database_url
+mbq.metrics
```
Comment (mattbriancon) [inline requirements-dev.txt]: nit: keep this list sorted.

File: docker-compose.yml [hunk 7]
```diff
@@ -20,6 +23,14 @@ services:
...
       - .:/app
+    links:
+      - postgres
     environment:
       PYTHON_VERSION: py37
       TOX_WORK_DIR: /tox
+      DATABASE_URL: 'postgres://postgres:postgres@postgres:5432/pubsub-pg'
+
+  postgres:
+    image: postgres:9.6.11
```
Comment (mattbriancon) [inline docker-compose.yml]: Let's pin to the minor version only. Our production DB's automatically get patch version updates.

File: tests/test_settings.py (hunk 5 above)
Comment (mattbriancon) [reply tests/test_settings.py]: I should have said: see `tests/conftest.py`. pytest automatically looks for that file.

File: mbq/pubsub/models.py (hunk 2 above)
Comment (tayloraburgess) [reply mbq/pubsub/models.py]: Intending for `timestamp` to be the timestamp on the original message itself & `error_timestamp` to be when the message is put on the DLQ, but realizing as I write this that we may not actually have that information (and even if so the field names should probably be more descriptive)

File: mbq/pubsub/models.py (hunk 3 above)
Comment (tayloraburgess) [reply mbq/pubsub/models.py]: Yeah, re: your comment above, I think I may just consolidate `message_timestamp` field & then use `created_at` for when the row is inserted

File: tests/test_settings.py (hunk 5 above)
Comment (tayloraburgess) [reply tests/test_settings.py]: Ahh I missed that entirely. I'll consolidate into there

**Comment by tayloraburgess** [comment]
@mattbriancon @Pastromhaug updated. Using `tests/conftest` actually just eliminated the need for everything I had set up in `test_settings`, and the Postgres docker-compose service as well (running migrations using SQLite in memory works fine).

Actually running `makemigrations` is a bit more convoluted with this setup, however--might be easier to discuss in person, but thinking we might be best to write a script for it

**Pull request merged**